integration/clean/main.go

```diff
@@ -0,0 +1,76 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent/integration/clean/ami"
+	"github.com/aws/amazon-cloudwatch-agent/integration/clean/dedicated_host"
+	"github.com/aws/amazon-cloudwatch-agent/integration/clean/ssm"
+)
+
+const (
+	defaultDaysToKeep = 5
+	cleanerTypeAll    = "all"
+	delimiter         = ","
+)
+
+var (
+	// flags
+	daysToKeep    int
+	cleanersToRun string
+
+	cleanerTypes   = []string{ssm.Type, ami.Type, dedicated_host.Type}
+	cleanerOptions = strings.Join(cleanerTypes, delimiter)
+)
+
+func init() {
+	flag.IntVar(&daysToKeep, "keep", defaultDaysToKeep, "Days to keep a resource before cleaning it.")
+	flag.StringVar(&cleanersToRun, "clean", cleanerTypeAll,
+		fmt.Sprintf("Determines which cleaners to run. (e.g. -clean ami%sssm)\nOptions: %s,%s", delimiter, cleanerTypeAll, cleanerOptions))
+}
+
+func main() {
+	flag.Parse()
+
+	ctx := context.Background()
+
+	keepDuration := -1 * time.Hour * 24 * time.Duration(daysToKeep)
+
+	if strings.Contains(cleanersToRun, cleanerTypeAll) {
+		cleanersToRun = cleanerOptions
+	}
+
+	// The date used to determine if a resource can be cleaned.
+	// Anything older than the date will be removed.
+	expirationDate := time.Now().UTC().Add(keepDuration)
+	log.Printf("Expiration date set to %v", expirationDate)
+
+	for _, cleaner := range strings.Split(cleanersToRun, delimiter) {
```

Comment:
I will have to think about this more. 

I am not sure I understand the value of consolidating all of this code into one main that executes the different cleaners. Also, you noted in the PR description `to avoid throttling limit for AWS account`, but I can't gleam from this PR how this addresses throttling limits.

Reply:
Initially I thought this would address the **throttling limit for the internal integration test**; however, changing the purpose to reduce the billing and also avoid reaching the quota limit for each corresponding service.

Reply:
Can you update the README for this at `integration/clean/README.md` to reflect the changes you are introducing to it?

Reply:
Yes nice catch. Have changed the READ ME and how to use the code with the variables.